### PR TITLE
Add support for more degrees of freedom and separate tests

### PR DESCRIPTION
**Pull request by PerezHz**
While working on #12, I tried to test easy examples in 2d and 3d (free particle, Kepler problem, etc.), following SICM as close as possible. So I tried to make those examples work; this PR is the fruit of those attempts. I made some changes in the `∂` function so that it now supports partially-differentiating vectorial functions; these changes were somewhat ugly, and I'm sure we can make it better, but for now, it gets the job done while exploiting Julia's multiple dispatch.

Also, this PR separates tests in several files.

### CODE REVIEW

File: test/eulerlagrange.jl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    function L_kepler(local_tuple::UpTuple)
+        t, q, qdot = local_tuple.data
+        r = (q[1]^2+q[2]^2)^(1/2)
+        (qdot[1]^2+qdot[2]^2)*(m/2)+μ/r
+    end
+
+    q = UpTuple([x, y])
+
+    # Equivalent to Eq. (1.48) of SICM
+    @test Lagrange_Equations(L_kepler)(q)(t).data == SymExpr[m * (D(D(x)))(t) + ((x)(t) ^ 2 + (y)(t) ^ 2) ^ -1.5 * (x)(t) * μ, m * (D(D(y)))(t) + ((x)(t) ^ 2 + (y)(t) ^ 2) ^ -1.5 * (y)(t) * μ]
```
Comment (MasonProtter) [inline test/eulerlagrange.jl]: What does `SymExpr[a, b]` do here? it is the SICM equivalent of `(up a b)`? If so, perhaps we should use `up(a, b)`

**Review by MasonProtter — COMMENTED** [review]
Looks great, just one concern about a constructor but that's probably something I missed from your last PR.

File: test/eulerlagrange.jl (hunk 1 above)
Comment (PerezHz) [reply test/eulerlagrange.jl]: Thanks, I used that `SymExpr[...]` to construct a vector of `SymExpr` which then I used to construct an `UpTuple` and compare that to the equations of motion computed by `Lagrange_Equations`. But you're right, we should use `up`. The way it is now, I'm doing it both ways: the line
`kepler_eqs = [m * (D(D(x)))(t) ...`
constructs a `Vector{SymExpr}` with the equations of motion for the Kepler problem; then we are able to test the output of `Lagrange_Equations` with either
```julia
@test lagrange_eqs_kepler_t.data == kepler_eqs # comparing two `Vector{SymExpr}`s
```
or
```julia
@test lagrange_eqs_kepler_t == up(kepler_eqs) # comparing two `UpTuple`s
```
where `lagrange_eqs_kepler_t` is an `UpTuple`

**Comment by PerezHz** [comment]
appveyor was failing on x86 builds (32-bit platforms), and the error was rather obscure, so I just turned it off for those cases

**Comment by MasonProtter** [comment]
Awesome! Great work, I’ve gotta catch up!

**Pull request merged**